examples/load_and_upsample_bboxes.py

```diff
@@ -0,0 +1,395 @@
+
+    # plot box at this frame
+    top_left_corner = (
+        ds.position.sel(time=frame_idx).data
+        - ds.shape.sel(time=frame_idx).data / 2
+    ).squeeze()
+
+    bbox = plt.Rectangle(
+        xy=tuple(top_left_corner),
+        width=ds.shape.sel(time=frame_idx, space="x").item(),
+        height=ds.shape.sel(time=frame_idx, space="y").item(),
+        edgecolor="red",
+        facecolor="none",
+        linewidth=1.5,
+        label="current frame",
+    )
+    ax.add_patch(bbox)
+
+    # plot box's centroid at this frame with red ring
+    ax.scatter(
+        x=ds.position.sel(time=frame_idx, space="x"),
+        y=ds.position.sel(time=frame_idx, space="y"),
+        s=15,
+        color="red",
+    )
+
+    # plot past centroid positions in blue
+    if frame_idx > 0:
+        ax.scatter(
+            x=ds.position.sel(time=slice(0, frame_idx - 1), space="x"),
+            y=ds.position.sel(time=slice(0, frame_idx - 1), space="y"),
+            s=10,
+            color="tab:blue",
+            label="past frames",
+        )
+
+    # plot future centroid positions in white
+    ax.scatter(
+        x=ds.position.sel(time=slice(frame_idx + 1, end_frame_idx), space="x"),
+        y=ds.position.sel(time=slice(frame_idx + 1, end_frame_idx), space="y"),
+        s=10,
+        color="white",
+        label="future frames",
+    )
+
+    # set title and labels
+    ax.set_title(f"Frame {frame_idx}")
+    ax.set_xlabel("x (pixles)")
+    ax.set_ylabel("y (pixels)")
+    ax.set_xlabel("")
+    if frame_idx == 1:
+        ax.legend()
+
+fig.tight_layout()
+
+# %%
+# We used ``xarray``'s ``.sel()`` method to select the data for the
+# relevant frames directly.
+#
+# The centroid at each frame is marked with a red marker. The past centroid
+# positions are shown in blue and the future centroid positions in white.
+# Note that in this case the camera is not static relative to the environment.
+
+# %%
+# Fill in empty values with forward filling
+# ----------------------------------------------------
+# We can fill in the frames with empty values for the  ``position`` and
+# ``shape`` arrays by taking the last valid value in time. In this way, a
+# box's position and shape stay constant if for a current frame the box
+# has no annotation defined.
+
+ds_ff = ds.reindex(
+    {"time": list(range(ds.time[-1].item()))},
+    method="ffill",  # propagate last valid index value forward
+)
+
+# %%
+# We can verify with a plot that the empty values have been filled in
+# using the last valid value in time.
+#
+# For this we define a convenience function to plot the x and y coordinates
+# of the ``position`` and  ``shape`` arrays, for the input dataset and for
+# a filled one.
+
+
+def plot_position_and_shape_xy_coords(ds_input_data, ds_filled, color_filled):
+    """Compare the x and y coordinates of the position and shape arrays in time
+    for the input and filled datasets.
+    """
+    _, axs = plt.subplots(2, 2, figsize=(8, 6))
```

Comment:
Let's make the figure a bit wider, on my screen there is barely any horizontal white space between the subplots.
```suggestion
    _, axs = plt.subplots(2, 2, figsize=(9, 6))
```

Reply:
or alternatively use `plt.tight_layout()`

Reply:
Here's a screenshot so you know what I mean: 

![Screenshot 2024-12-05 at 18 24 19](https://github.com/user-attachments/assets/0ac298cf-f81a-4f01-8cb8-022ba24067d0)